ffcx/ir/elementtables.py

```diff
@@ -500,9 +520,11 @@ def build_optimized_tables(
                 d = local_derivatives[i]
                 sub_tbl = j.tabulate(d, pts)[d]
                 sub_tbl = sub_tbl.reshape(1, 1, sub_tbl.shape[0], sub_tbl.shape[1])
-                for i in all_tensor_factors:
-                    if i.values.shape == sub_tbl.shape and np.allclose(i.values, sub_tbl):
-                        tensor_factors.append(i)
+                for tensor_factor in all_tensor_factors:
```

Comment:
@IgorBaratta Here `i` was redeclared (from line 5.18 enumerate), I assume that was a bug.
Do you agree?

Reply:
Yes, thanks for the fix